### ISSUE TITLE
Support for resourceID in ComputeNetworkEndpointGroup

### ISSUE THREAD

**Issue by erikjoh** [GoogleCloudPlatform/k8s-config-connector#369]
Hi!

We have a potential use case for creating multiple [Network Endpoint Group](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/crds/compute_v1beta1_computenetworkendpointgroup.yaml) resources with the same resource name under different compute zones in the same project/namespace. I think the only way this can be done is if the resource supported the `resourceId` field. For example:

```yaml
apiVersion: compute.cnrm.cloud.google.com/v1beta1
kind: ComputeNetworkEndpointGroup
metadata:
  name: someservice-neg-1  # different
  namespace: somenamespace
spec:
  resourceID: someservice-neg  # same 
  networkEndpointType: GCE_VM_IP_PORT
  networkRef:
    name: somenetwork
  subnetworkRef:
    name: somesubnet
  location: europe-west1-b  # different
---
apiVersion: compute.cnrm.cloud.google.com/v1beta1
kind: ComputeNetworkEndpointGroup
metadata:
  name: someservice-neg-2  # different
  namespace: somenamespace
spec:
  resourceID: someservice-neg  # same
  networkEndpointType: GCE_VM_IP_PORT
  networkRef:
    name: somenetwork
  subnetworkRef:
    name: somesubnet
  location: europe-west1-c  # different
```

Currently low priority.

**Comment by spew** [comment]
Hi @erikjoh thanks for opening this, we will be adding support for resourceID to ComputeNetworkEndpointGroup soon.

Another way this could be done is by placing the K8s resources into different namespaces (so they can legally have the same K8s name), but using the same project-id annotation on both.

**Comment by maqiuyujoyce** [comment]
Hi @erikjoh , we just released [v1.36.0](https://github.com/GoogleCloudPlatform/k8s-config-connector/releases/tag/1.36.0), which supports the resourceID field in `ComputeNetworkEndpointGroup`.
